src/language/templates/ast/ast.cpp

```diff
@@ -16,6 +16,51 @@
 namespace nmodl {
 namespace ast {
 
+///
+///  Ast member function definition
+///
+
+Ast *Ast::clone() const { throw std::logic_error("clone not implemented"); }
+
+std::string Ast::get_node_name() const {
+  throw std::logic_error("get_node_name() not implemented");
+}
+
+std::shared_ptr<StatementBlock> Ast::get_statement_block() const {
+  throw std::runtime_error("get_statement_block not implemented");
+}
+
+const ModToken *Ast::get_token() const { return nullptr; }
+
+symtab::SymbolTable *Ast::get_symbol_table() {
+  throw std::runtime_error("get_symbol_table not implemented");
+}
+
+void Ast::set_symbol_table(symtab::SymbolTable * /*symtab*/) {
+  throw std::runtime_error("set_symbol_table not implemented");
+}
+
+void Ast::set_name(const std::string & /*name*/) {
+  throw std::runtime_error("set_name not implemented");
+}
+
+void Ast::negate() { throw std::runtime_error("negate not implemented"); }
```

Comment:
just to understand: it's true that methods are put in cpp file instead of header to avoid re-compilation upon modification. Above functions were moved to .cpp for the same reason? Or to also reduce the header size or something? 

(functions like above don't have any implementation logic and hence I put them initially in .hpp).

Reply:
I put the class member definitions in .cpp only to mitigate the compilation time of all the files including ast.hpp. Most of them are virtual methods so there is no performance gain to put their implementation in .hpp